pandas/tests/io/excel/test_writers.py

```diff
@@ -1549,3 +1550,27 @@ def test_subclass_attr(klass):
     attrs_base = {name for name in dir(ExcelWriter) if not name.startswith("_")}
     attrs_klass = {name for name in dir(klass) if not name.startswith("_")}
     assert not attrs_base.symmetric_difference(attrs_klass)
+
+
+@pytest.mark.parametrize("merge_cells", [True, False])
+def test_excel_round_trip_with_periodindex(merge_cells, engine):
+    # GH#60099
+    df = DataFrame(
+        {"A": [1, 2]},
+        index=MultiIndex.from_arrays(
+            [period_range("2023-01", "2023-02", freq="M"), ["X", "Y"]],
+            names=["date", "category"],
+        ),
+    )
+
+    with BytesIO() as buffer:
```

Comment:
An interesting approach and not writing to disk has merit, but for now I think we should prefer to have consistency in the tests. Can you add this as a method to the class `TestRoundTrip` above and follow the pattern of using `tmp_excel` there.

Reply:
Sorry for the trouble. I'll modify the test.